app/src/main/java/ca/carleton/gcrc/n2android_mobile1/cordova/PluginActions.java

```diff
@@ -103,25 +107,19 @@ public void couchbaseCreateDocument(final JSONObject doc, final CallbackContext
             final DocumentDb docDb = getDocumentDb();
 
             try {
-
                 fusedLocationProviderClient.getLastLocation()
                         .addOnSuccessListener(new OnSuccessListener<Location>() {
                             @Override
                             public void onSuccess(Location location) {
                                 if (location != null) {
                                     Log.v(TAG, location.toString());
 
-                                    try {
-                                        double latitude = location.getLatitude();
-                                        double longitude = location.getLongitude();
-
-                                        JSONObject locationObject = new JSONObject();
-                                        locationObject.put("wkt", "MULTIPOINT((" + longitude + " " + latitude + "))");
-                                        locationObject.put("nunaliit_type", "geometry");
-                                        doc.put("nunaliit_geom", locationObject);
-                                    } catch (JSONException je) {
-                                        // If you can't get a location, carry on.
+                                    if (doc.has("nunaliit_mobile_needs_new_location") ||
```

Comment:
According to this logic, if i set `nunaliit_mobile_needs_new_location` to `false` it will still add the location. Is this what we want? We could unset the variable in javascript but it's not as intuitive as just doing a `doc.get()` here. What do you think?